authority/tls.go

```diff
@@ -352,6 +348,19 @@ func (a *Authority) Rekey(oldCert *x509.Certificate, pk crypto.PublicKey) ([]*x5
 		newCert.ExtraExtensions = append(newCert.ExtraExtensions, ext)
 	}
 
+	// Check if the certificate is allowed to be renewed, policies or
+	// constraints might change over time.
+	if err := a.isAllowedToSignX509Certificate(newCert); err != nil {
```

Comment:
One of the things discussed when implementing allow/deny was the potential for outages when an updated policy doesn't allow the certificate to be renewed. That's why policies were not evaluated on renewal. Enabling it here may need to be reconsidered.

Reply:
I can remove the policy and only consider the constraints here. Sounds that ok?

Reply:
Fixed with 2eba5326